src/goto-programs/goto_convert.cpp

```diff
@@ -1363,20 +1329,14 @@ void goto_convertt::convert_return(
   goto_programt &dest,
   const irep_idt &mode)
 {
-  if(!targets.return_set)
-  {
-    error().source_location=code.find_source_location();
-    error() << "return without target" << eom;
-    throw 0;
-  }
+  DATA_INVARIANT(
+    targets.return_set,
+    code.find_source_location().as_string() + ": return without target");
 
-  if(!code.operands().empty() &&
-     code.operands().size()!=1)
-  {
-    error().source_location=code.find_source_location();
-    error() << "return takes none or one operand" << eom;
-    throw 0;
-  }
+  DATA_INVARIANT(
+    code.operands().empty() || code.operands().size() == 1,
+    code.find_source_location().as_string() +
+      ": return takes none or one operand");
```

Comment:
I think this should be covered in `std_code.h`, though it seems the checks in there need amending.

Reply:
I think it's fine to have "redundant" invariant checks as long as the following code actually requires the condition holds.